### PR TITLE
Added Basic Functional tests

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPMedia/Tests/FunctionalTests/Scenarios/AdPlayback.swift [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+            MediaHit(eventType: EventType.AD_COMPLETE, playhead: 0, ts: 55000),
+            MediaHit(eventType: EventType.ADBREAK_COMPLETE, playhead: 0, ts: 55000),
+            MediaHit(eventType: EventType.PLAY, playhead: 0, ts: 55000),
+            MediaHit(eventType: EventType.PLAY, playhead: 1, ts: 56000),
+            MediaHit(eventType: EventType.PING, playhead: 51, ts: 106000),
+            MediaHit(eventType: EventType.SESSION_COMPLETE, playhead: 55, ts: 110000)
+        ]
+
+        //verify
+        checkHits(expectedHits: expectedHits)
```
Comment (praveek) [inline AEPMedia/Tests/FunctionalTests/Scenarios/AdPlayback.swift:175]: Can we add a test case for 1 sec tracking ?

File: AEPMedia/Tests/FunctionalTests/Scenarios/ChapterPlayback.swift [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            MediaHit(eventType: EventType.PING, playhead: 11, ts: 11000),
+            MediaHit(eventType: EventType.CHAPTER_COMPLETE, playhead: 15, ts: 15000),
+            MediaHit(eventType: EventType.SESSION_COMPLETE, playhead: 15, ts: 15000)
+        ]
+
+        //verify
+        checkHits(expectedHits: expectedHits)
+    }
+
+    func testPrerollAd_DownloadedTracker() {
```
Comment (praveek) [inline AEPMedia/Tests/FunctionalTests/Scenarios/ChapterPlayback.swift:104]: Typo

File: AEPMedia/Tests/FunctionalTests/Scenarios/AdPlayback.swift (hunk 1 above)
Comment (addb) [reply AEPMedia/Tests/FunctionalTests/Scenarios/AdPlayback.swift:175]: Yes I am adding more cases.

File: AEPMedia/Tests/FunctionalTests/Scenarios/ChapterPlayback.swift (hunk 2 above)
Comment (addb) [reply AEPMedia/Tests/FunctionalTests/Scenarios/ChapterPlayback.swift:104]: yes

**Pull request merged**